internal/engine/compiler/cpuid_amd64.s

```diff
@@ -0,0 +1,14 @@
+#include "textflag.h"
+
+// lifted from github.com/intel-go/cpuid and src/internal/cpu/cpu_x86.s
+// func cpuid(arg1, arg2 uint32) (eax, ebx, ecx, edx uint32)
+TEXT ·cpuid(SB), NOSPLIT, $0-24
+	MOVL arg1+0(FP), AX
+	MOVL arg2+4(FP), CX
+	CPUID
+	MOVL AX, eax+8(FP)
+	MOVL BX, ebx+12(FP)
```

Comment:
do we need AX and BX register values? I think we could just delete them if you don't use.
```suggestion

TEXT ·cpuid(SB), NOSPLIT, $0-16
	MOVL arg1+0(FP), AX
	MOVL arg2+4(FP), CX
	CPUID
```

Reply:
Other feature queries might use all 4 (e.g. BMI uses BX).

Reply:
Other feature queries might use all 4 (e.g. BMI uses BX).

Reply:
Also, in theory (maybe not in practice) you should call `cpuid(0, 0)` and inspect `EAX` to figure out if you can trust `cpuid(1, 0)`, and similarly call `cpuid(0x8000_0000, 0)` before `cpuid(0x8000_0001, 0)`.

https://www.amd.com/system/files/TechDocs/25481.pdf
> The smallest function number of the standard function range is Fn0000_0000. The largest function number of the standard function range, for a particular implementation, is returned in CPUID
Fn0000_0000_EAX.
> The smallest function number of the extended function range is Fn8000_0000. The largest function number of the extended function range, for a particular implementation, is returned in CPUID
Fn8000_0000_EAX.

Reply:
cool, thanks for the info!